packages/framework/src/Controller/Admin/SalesRepresentativeController.php

```diff
@@ -166,4 +157,56 @@ public function deleteAction(int $id): Response
 
         return $this->redirectToRoute('admin_salesrepresentative_list');
     }
+
+    /**
+     * @param int $id
+     */
+    #[Route(path: '/sales-representative/delete-confirm/{id}', requirements: ['id' => '\d+'])]
+    public function deleteConfirmAction(int $id)
+    {
+        try {
+            $salesRepresentative = $this->salesRepresentativeFacade->getById($id);
+            $customersUsingThisSalesRepresentative = $this->customerUserFacade->findEmailsOfCustomerUsersUsingSalesRepresentative($id);
+            $customersCount = count($customersUsingThisSalesRepresentative);
+
+            if ($customersCount > 0) {
+                $customersEnumeration = implode('<br>', array_slice($customersUsingThisSalesRepresentative, 0, 20));
+
+                $message = t(
+                    'Sales representative "%label%" is assigned to %count% customers and will be removed if you proceed.<br><br>
+                            Customers:<br>
+                            %customersEnumeration%<br>
+                            %extraCount%
+                            Do you really want to remove sales representative "%label%" permanently?',
+                    [
+                        '%label%' => $salesRepresentative->getPresentationalLabel(),
+                        '%count%' => $customersCount,
+                        '%extraCount%' => $customersCount - 20,
```

Comment:
what is this extra count good for? It renders just as a number, was that intended? (I would also extract the number 20 into a constant)

Reply:
i consulted this with PO and this was the solution we came up with

it is used in confirm popup window when we want to delete sales representative and if they are assigned to some customers we list them, but only up to 20 and in case if there is more of them then we list those first 20 and add the count as additional information for the others